### PR TITLE
windows_ad_join resource - add newname property

### PR DESCRIPTION
**Pull request by derekgroh**
Signed-off-by: Derek Groh <derekgroh@github.io>

### Description

Adds the NewName parameter for the Add-Computer command

https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.management/add-computer?view=powershell-5.1#optional-parameters

### Issues Resolved

Stack Overflow request to be able to rename a computer when joining the domain.

### Check List

- [ ] New functionality includes tests (is there a good way to test new properties?)
- [X] All tests pass
- [X] RELEASE\_NOTES.md has been updated if required (not required for bugfixes, required for API changes)
- [X] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

File: lib/chef/resource/windows_ad_join.rb [hunk 1]
```diff
@@ -52,6 +52,9 @@ class WindowsAdJoin < Chef::Resource
                description: "Controls the system reboot behavior post domain joining. Reboot immediately, after the Chef run completes, or never. Note that a reboot is necessary for changes to take effect.",
                default: :immediate
 
+      property :newname, String,
```
Comment (stuartpreston) [inline lib/chef/resource/windows_ad_join.rb]: Please can we follow the Ruby convention here, i.e. rename to `:new_name`

File: lib/chef/resource/windows_ad_join.rb [hunk 2]
```diff
@@ -64,6 +67,7 @@ class WindowsAdJoin < Chef::Resource
           cmd << "$credential = New-Object System.Management.Automation.PSCredential (\"#{new_resource.domain_user}\",$pswd);"
           cmd << "Add-Computer -DomainName #{new_resource.domain_name} -Credential $credential"
           cmd << " -OUPath \"#{new_resource.ou_path}\"" if new_resource.ou_path
+          cmd << " -NewName \"#{new_resource.newname}\"" if new_resource.newname
```
Comment (stuartpreston) [inline lib/chef/resource/windows_ad_join.rb]: Please use `:new_name` / `new_resource.new_name`

File: lib/chef/resource/windows_ad_join.rb (hunk 1 above)
Comment (derekgroh) [reply lib/chef/resource/windows_ad_join.rb]: right we discussed this, sorry windows habits

**Comment by stuartpreston** [comment]
Thanks for your contribution!

**Pull request merged**